crates/kubelet/src/kubelet.rs

```diff
@@ -117,14 +111,16 @@ impl<P: Provider> Kubelet<P> {
         };
         let mut operator_runtime = OperatorRuntime::new(&self.kube_config, operator, Some(params));
         let operator_task = operator_runtime.start().fuse().boxed();
-
         // These must all be running for graceful shutdown. An error here exits ungracefully.
         let core = Box::pin(async {
             tokio::select! {
-                res = signal_handler => res.map_err(|e| {
-                    error!("Signal handler task joined with error {:?}", &e);
-                    e
-                }),
+                res = signal_handler => match res {
```

Comment:
I think its weird to have a task that just waits and does the actual work here. We are also implicitly requiring that none of the other arms of this `select!` will ever complete first. 

Can we run `provider.shutdown()` after the `try_join!` on 134?.

Reply:
I wondered about that as well but figured it was maybe in order to be able to add additional shutdown handling down the line.
Splitting this out into a handler communicating via a signal would allow implementing additional handlers that watch for a shutdown CRD, look for local files, offer a rest API or other stuff fairly easily. 
Just start an additional task with the new handling that sets the shutdown AtomicBool to true.

Come to think of it, we could allow providers to define a closure that can handle custom shutdown triggers and spin up a task running this in the Kubelet.  I'm not advocating for this, just a thought that came to me pretty much while writing this :)